fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/PercentileArray.java

```diff
@@ -74,6 +75,14 @@ public PercentileArray(boolean distinct, Expression arg0, Expression arg1) {
         super("percentile_array", distinct, arg0, arg1);
     }
 
+    @Override
+    public void checkLegalityBeforeTypeCoercion() {
+        if (!getArgument(1).isConstant()) {
```

Comment:
If the limit is constant, can we also add the limit of not null and [0,1] value range?

Reply:
now FE can't sure the const fold ready in FE, so union check in BE